### ISSUE TITLE
Linode UI bypasses meta-proxy 

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#21838]
From @kinarashah on https://github.com/rancher/rancher/issues/21300#issuecomment-516198323

The UI code for Linode doesn't follow the same logic for validation and fetching data as other builtin drivers. https://github.com/rancher/ui/blob/master/lib/global-admin/addon/components/cru-cloud-credential/component.js#L188  
It needs to go through the `/meta/proxy` link, this will allow: 

- The whitelisted domains mentioned for the driver get respected 
- We can make use of cloud credentials support instead of actual passing of secret data from ui to linode api. 

Also need to fix, 
- On update/edit of node template, right now linode requires entering credential every time but this shouldn't if we leverage cloud credential (I could see it's missing `finishAndSelectCloudCredential` which other drivers have. @westlywright maybe you can point out exactly . where this change needs to happen, node template has the correct credID but it doesn't load in UI.) 

The current UI works, it just doesn't leverage meta-proxy the way we recommend + causes minor issues

**Comment by davidnuzik** [comment]
Version: master-head (v2.3) (8/14/19)

I performed the frontend checks in the UI. I can see now that when I create a Linode cloudCredential we go thru /meta/proxy/ to validate our credentials. This is covered further here: https://github.com/rancher/rancher/issues/22128#issuecomment-521416577

I checked nodeTemplate creation and I can see the linode api calls now go thru /meta/proxy (i.e. for regions, images, types, etc).
/meta/proxy/api.linode.com/v4/regions
/meta/proxy/api.linode.com/v4/images
/meta/proxy/api.linode.com/v4/linode/types

I created various nodeTemplates for Linode and smoke tested. I wasn't able to identify any issues. I tried various settings, set root pwd url, etc non-default options just for a sanity check (even though this area is not impacted; the important change is how we make the linode api calls now via /meta/proxy in the UI).
